Added bomi icon

schyzophrene-asynchrone opened the pull request:
My first try for a pull request.
Please tell me if I do anything wrong.

Commit:
Added bomi icon

luebking commented:
The commit contains a "gnome-shape" icon but links a "bomi" and a "calibre-gui" icon in alias.txt

schyzophrene-asynchrone commented:
The calibre-gui icon was already in your alias (I just added a newline).
I removed the gnome icon and replaced it with the bomi icon. It should be good now.

Sorry ^^

luebking commented:
> The calibre-gui icon was already in your alias 
> Indeed, my bad.
> Please squash the commits (and either do a force push or pick the result into your master and create a pull reqest from there)

schyzophrene-asynchrone commented:
How do I "squash commits" or "force push" ?

luebking commented:
git rebase -i HEAD~3

This will open an editor (iirc the default editor is vim*) with the last three commits.
You can reorganize commits here simply by juggling lines (should not be required in this case) and hint what to do with them. The default is "pick", if you replace that (do NOT touch commit messages or even hashes!) for the lower to commits (the newer one) to "s" (or "squash"), they'll be squashed into the next picked one. Saving the file and closing the editor will start some git action and then open a new editor where you can adjust the commit message of the new, combined, commit.

Finally, you can
   git push origin +bomi
to forcefully push the bomi branch (overriding the remote version of your git repo) - !notice! the "+" (otherwise you'll get a warning of a conflict and instruction to pull - what you do _not_ want to do)

In general, force pushing is discouraged.
It's semi-ok for work branches (like here), but an absolute no-go for "public" branches (notably "master") because it will cause much work for every other clone (to re-align commit history)
- if you're not comfortable with vim, you can eg.
  git config --global core.editor nano
  vim however has several advantages for git; not only syntax highlighting, but also vimdiff, which allows easier commit conflict resolution (but that's not relevant in the current situation)
  I assume even kwrite would work, but have never tested.

PS:
in case you spent the last 12 days on "how to f******\* git!!!": Just ask =)

schyzophrene-asynchrone commented:
> Finally, you can
>    git push origin +bomi

Just done, I hope it worked ^^

> - if you're not comfortable with vim, you can eg.
>   git config --global core.editor nano

I'm trying to learn vim, so thats' fine (I just have a few difficulties to copy/paste)

> PS:
> in case you spent the last 12 days on "how to f******\* git!!!": Just ask =)
> I was working on something else, and I just remembered I had to squash my commits (and git squash --help didn't return anything ^^)
> Thanks for your help, anyway !
